### ISSUE TITLE
Region file corresponding to hard-wired regions

### ISSUE THREAD

**Issue by q-rai** [MPAS-Dev/geometric_features#83]
Hi!

As part of my work on adding and testing region file support to some analysis members, I've created geojson files for the hard-wired regions that are used for some analysis members (e.g. Water Mass Census, and Surface Area Weighted Averages).

I propose to add them to this repository, I'm just not sure where exactly and under which name.

Currently, this is a region group containing:
- Arctic: lat >= 60
- Equatorial: -15 <= lat <= 15
- Southern Ocean: lat <= -50
- Nino 3: -5 <= lat <= 5; 210 <= lon <= 270
- Nino 4: -5 <= lat <= 5; 160 <= lon <= 210
- Nino 3.4: -5 <= lat <= 5; 190 <= lon <= 240
- Global

[This is what they look like](https://gist.github.com/q-rai/59ef3e4310e97f8125c14112d7a4e41f)

I can:
- Add the entire region group (I propose to put that in `ocean/region` as `Hard-Wired_Regions.geojson`); **or**
- Add each region individually, as it's already the case for other regions.

.. or something entirely different; I'll do whatever would be most useful.

Let me know what you prefer!

**Comment by xylar** [comment]
Hi @q-rai, the right way to do this would be the following:
1. put your `OceanRegions.geojson` file in the root of a new branch of `geometric_features`.
2. rename each features to have a unique, descriptive suffix.  For example, your Arctic region needs to be distinct from the `Arctic Ocean` region we already have, and the same for the `Southern Ocean`.  I don't know what the right suffix might be.  It needs to be more descriptive of what it will be used for than, for example, 'Region' or 'Hard-Wired'.  I thought of 'Analysis' but I'm not sure it's clear what the distinction between `Southern Ocean` from `Southern Ocean Analysis` would be...
3. Give the same tag to all the features (something not used for other features).  This will allow you to regroup the features into a single feature file later on.  You can do this with:
`./tag_features.py -f OceanRegions.geojson -t <tag> -o TaggedOceanRegions.geojson`
The tag will presumably be something similar to the suffix you add to all the feature names in 1.
4. Run `./split_features.py -f TaggedOceanRegions.geojson` to put the features into their proper place.
5. use `git add`, `git commit`, etc. to add all the split features to the repo as normal.

Later, when you wont to recreate your single file with all regions, you would do something like:
```
./merge_features.py -d ocean/region/ -t <tag> -o OceanRegions.geojson
./set_group_name.py -f OceanRegions.geojson -g <groupName>
```
The group name might be something like `AnalysisRegionsGroup`.  I think having a group name is helpful when you're using the masks later in MPAS, but I'm a little rusty.

**Comment by q-rai** [comment]
Thanks for the detailed response, @xylar!

As for the suffix, maybe @toddringler or @mark-petersen have a suggestion?
I could imagine something along the lines of 'rectangular' as that is something these regions have in common, and which differentiates them from the other regions available.

**Comment by xylar** [comment]
@q-rai, my preference would be for a suffix that somehow distinguishes their intended use from other region masks.  For example, we currently have a driver script for creating MOC (meridional overturning circulation) regions as a collection of other regions, and we give them a suffix (or maybe a prefix, I don't remember for sure) of MOC.  I realize these features are more general because they're used in many different analysis members, which makes them hard to name.

**Comment by q-rai** [comment]
I just double checked. These regions are used for
- Layer Volume Weighted Averages
- Surface Area Weighted Averages
- Water Mass Census

I could name it after one of them? The MOC files are also used for MHT so I figure that wouldn't be an issue.

**Comment by milenaveneziani** [comment]
@q-rai, @xylar: I think what we actually want for those AM's is the following:
1) compute regional analysis for all 3 of them on the ocean basin regions. These are *not* the currently hard-wired regions, but rather the ones that are created with driver_scripts/setup_ocean_basins.py.
So my preference would be to not have geojson files for the currently hardwired ocean basins (Arctic and Southern Ocean).
2) compute surface area weighted averages also for the Nino3.4 regions, for which we could use the geojson files that you created.
3) finally we can decide what other regional computation we want to have in these 3 AM's (the Equatorial region is one example), but we definitely want 1) and 2) as a given.

Does this make sense?

**Comment by xylar** [comment]
Ah, thank for the clarification, @milenaveneziani.  It sounds like we would need to be careful, then, to integrate these new regions (Nino3.4 and Equatorial) into the driver script that creates the MOC basins.

Also, if the MOC basins are used for more than just the MOC, maybe they should be renamed?

@milenaveneziani and @q-rai, once you have a definitive geojson file with the new regions you'd like to add, I can take care of tagging them, renaming them, splitting them into the appropriate locations in 'ocean/regions', modifying the driver script, etc.  Just let me know what you'd like.

**Comment by milenaveneziani** [comment]
@xylar: why would we want to integrate the Nino and Equatorial regions into the driver script that creates the MOC basins? Those regions look quite independent to me.
My preference would be to create the following groups:
1) MOC basins (true, they are used for the regional MHT calculation as well, but MOCMHTbasins doesn't sound too good to my ears..)
2) Ocean basins (perhaps these could include the Equatorial region)
3) Nino regions

And then MOC and MHT AM would use 1); the 2 am's listed above would use 2), plus the surfaceAvg am would use 3) as well.
Is this possible, do you think, @q-rai?

**Comment by xylar** [comment]
Ah, okay.  So I guess I misunderstood.  But the same driver that creates the MOC basins and ocean basins would also group the El Niño regions into a combined file with a tag, correct?  So the same driver script would be altered, just slightly differently than what I suggested.

**Comment by milenaveneziani** [comment]
oh, yes, that sounds good. Should we rename the driver maybe, to something like create_oceanRegion_groups?

**Comment by xylar** [comment]
@milenaveneziani, so are we agreed that all ocean regions are already covered except the following?
* Equatorial region (-15 to 15), which I will name `Global Ocean 15N to 15S` to be consistent with similar regions already in the database
* Nino 3
* Nino 4
* Nino 3.4

>  Should we rename the driver maybe, to something like create_oceanRegion_groups?
Maybe `create_ocean_region_groups` is a cleaner way of handling case and underscores?

I assume we would call the new group `NinoBasinRegionsGroup` to be consistent with `OceanBasinRegionsGroup` and `MOCBasinRegionsGroup`?

**Comment by xylar** [comment]
Actually referring to "El Niño Basins" doesn't really make sense.  Should I just call it `NinoRegionsGroup`?  And tag each feature for that group with `Nino` or maybe `Nino_Region`?

**Comment by xylar** [comment]
I think #84 should address this issue.  Let me know what we think?

**Comment by milenaveneziani** [comment]
sounds good to me @xylar. Reviewing #84 now.

**Comment by milenaveneziani** [comment]
oh, one thing I forgot to ask before reviewing #84: does it make sense to add the Global ocean to the oceanBasinRegionGroup?
I know that the MOC and MHT am's would not benefit from this (because the global calculation is done separately), but wouldn't it be better to do that for the layerWeightedAvgs, the surfaceAvgs and the waterMassCensus AM's? What do you think @q-rai?

**Comment by q-rai** [comment]
@milenaveneziani MHT is already computing the global MHT alongside (optional) regional MHT, so it wouldn't need an additional region for this. As far as I'm aware, MOC works the same way.
So, I agree that a global region is only required for layerWeightedAvgs, the surfaceAvgs and the waterMassCensus, none of which have a stand-alone global version.

**Comment by milenaveneziani** [comment]
yes, that's what I thought, thanks for the feedback @q-rai. So, @xylar: how about we add the globalOceanBasin to the OceanBasinRegionGroup in #86?

**Comment by q-rai** [comment]
@xylar @milenaveneziani Yes, I was just in the process of proposing that over in #84!

**Comment by xylar** [comment]
@q-rai and @milenaveneziani, was this addressed back in February by #84?

**Comment by xylar** [comment]
I think so, so I'm going to close.
